rust/src/data_structures.rs

```diff
@@ -389,6 +477,93 @@ pub enum DtlsFingerprint {
     },
 }
 
+fn hex_as_bytes(input: &str, output: &mut [u8]) {
+    for (i, v) in output.iter_mut().enumerate() {
+        *v = u8::from_str_radix(&input[i * 3..(i * 3) + 2], 16)
+            .map_err(|error| {
+                panic!("Failed to parse value {input} as series of hex bytes: {error}")
+            })
+            .unwrap();
+    }
+}
+
+impl DtlsFingerprint {
+    pub(crate) fn to_fbs(self) -> web_rtc_transport::Fingerprint {
+        match self {
+            DtlsFingerprint::Sha1 { .. } => web_rtc_transport::Fingerprint {
+                algorithm: web_rtc_transport::FingerprintAlgorithm::Sha1,
+                value: self.value_string(),
+            },
+            DtlsFingerprint::Sha224 { .. } => web_rtc_transport::Fingerprint {
+                algorithm: web_rtc_transport::FingerprintAlgorithm::Sha224,
+                value: self.value_string(),
+            },
+            DtlsFingerprint::Sha256 { .. } => web_rtc_transport::Fingerprint {
+                algorithm: web_rtc_transport::FingerprintAlgorithm::Sha256,
+                value: self.value_string(),
+            },
+            DtlsFingerprint::Sha384 { .. } => web_rtc_transport::Fingerprint {
+                algorithm: web_rtc_transport::FingerprintAlgorithm::Sha384,
+                value: self.value_string(),
+            },
+            DtlsFingerprint::Sha512 { .. } => web_rtc_transport::Fingerprint {
+                algorithm: web_rtc_transport::FingerprintAlgorithm::Sha512,
+                value: self.value_string(),
+            },
+        }
+    }
+
+    pub(crate) fn from_fbs(fingerprint: &web_rtc_transport::Fingerprint) -> DtlsFingerprint {
+        match fingerprint.algorithm {
+            web_rtc_transport::FingerprintAlgorithm::Sha1 => {
+                let mut value_result = [0_u8; 20];
+
+                hex_as_bytes(fingerprint.value.as_str(), &mut value_result);
```

Comment:
Why is it sent as hex and not binary from worker in the first place?

Reply:
This is probably because the client needs to use it in hex mode, so everyone in the chain, starting from worker, provides it in this form. I wouldn't be totally against having it in binary form in server side in a future.